File: JobTracker/JobTracker.Domain/Entities/Offer.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace JobTracker.Domain.Entities
{
    public class Offer
    {
        public int Id { get; private set; }
        public int ApplicationId { get; private set; }
        public decimal Salary { get; private set; }
        public DateTime OfferDate { get; private set; }
        public DateTime? Deadline { get; private set; }
        public string? Benefits { get; private set; }

        public JobApplication JobApplication { get; private set; }

        private Offer() { }
        public Offer(int applicationId, decimal salary, DateTime offerDate, DateTime? deadline, string? benefits)
        {
            ApplicationId = applicationId;
            Salary = salary;
            OfferDate = offerDate;
            Deadline = deadline;
            Benefits = benefits;
        }
    }
}

File: frontend/JobTracker/src/app/features/models/offer.ts
export interface Offer {
  id: number;
  applicationId: number;
  companyName: string;
  jobPosition: string;
  salary: number;
  offerDate: string;
  deadline: string;
  benefits?: string;
}

export interface OfferUpdate {
  id?: number;
  applicationId: number;
  salary: number;
  offerDate: string;
  deadline: string;
  benefits: string | null;
}
